wallet/src/actors/worker/methods.rs

```diff
@@ -777,7 +759,18 @@ impl Worker {
             .flatten()
             .map(|json| {
                 log::trace!("getBlock request result: {:?}", json);
-                serde_json::from_value::<types::ChainBlock>(json).map_err(node_error)
+                // Set confirmed to true if the result contains {"confirmed": true}
+                let mut confirmed = false;
+                if let Some(obj) = json.as_object() {
+                    if let Some(c) = obj.get("confirmed") {
+                        if let Some(true) = c.as_bool() {
+                            confirmed = true;
+                        }
+                    }
+                }
```

Comment:
After you last changes, block will always have a `confirmed` field, don't they?

Reply:
This will work as well, but should I rewrite it to return an error if there is no `confirmed` field?

Reply:
This way the wallet could detect if it connects to a node running an old version.